### PR TITLE
Jjnavanon/sucu 45 user get all users

### PR DESCRIPTION
**Pull request by JJnvn**
### Type of Change

- [x] New features
- [ ] Bug fixes
- [ ] Breaking changes
- [ ] Refactor
- [ ] Documentation updates
- [ ] Other (please specify):

## Description
- create new dto to use ( {"page" : ... , "limit" : } )
- get-all -users feature

### CODE REVIEW

File: internal/domain/usecases/user_usecase.go [hunk 1]
```diff
@@ -31,8 +31,28 @@ func NewUserUsecase(cfg config.Config, logger *zap.Logger, userRepository reposi
...
-func (u *userUsecase) GetAllUsers(req *dtos.UserDTO) (*[]dtos.UserDTO, *apperror.AppError) {
-	return nil, nil
+func (u *userUsecase) GetAllUsers(req *dtos.GetAllUsersDTO) (*[]dtos.UserDTO, *apperror.AppError) {
+	offset := req.Limit * (req.Page - 1)
+	limit := req.Limit
+
+	users, err := u.userRepository.FindAllUsers(limit, offset)
+	if err != nil {
+		u.logger.Named("GetAllUsers").Error(constant.ErrUserNotFound, zap.Error(err))
+		return nil, apperror.BadRequestError(constant.ErrUserNotFound)
```
Comment (bookpanda) [inline internal/domain/usecases/user_usecase.go]: imho, let's use internal server error instead bc we would've checked for bad request errors in handlers layer already. So if an error's gonna occur in repo, it's more likely to be 500 i think

File: internal/interface/handlers/user_handler.go [hunk 2]
```diff
@@ -30,7 +30,19 @@ func NewUserHandler(userUsecase usecases.UserUsecase, validator validator.DTOVal
...
 // @Router /users [get]
 func (h *UserHandler) GetAllUsers(c *fiber.Ctx) error {
-	return nil
+	var req dtos.GetAllUsersDTO
+	if err := c.BodyParser(&req); err != nil {
+		resp := response.NewResponseFactory(response.ERROR, err.Error())
+		return resp.SendResponse(c, fiber.StatusBadRequest)
+	}
+
+	resReturn, err := h.userUsecase.GetAllUsers(&req)
```
Comment (bookpanda) [inline internal/interface/handlers/user_handler.go:45]: let's add some checks here for negative values in page, limits

Comment (bookpanda) [reply internal/interface/handlers/user_handler.go:45]: e.g. page = -1 => bad request

**Review by bookpanda — COMMENTED** [review]
lgtm, only minor changes

**Review by flagrantii — APPROVED** [review]

**Pull request merged**